### PR TITLE
feat(ng-add): respect project default inlineStyle, inlineTemplate and spec option value

### LINKED ISSUE #11874: Material schematics doesn't work with configured angular schematics (scss, spec, etc...)
#### Bug, feature request, or proposal:
BUG

#### What is the expected behavior?
When I've configured the CLI to use scss and skip tests for generated components, I would expect the material schematics to use already defined configuration when generating components (such as the dashboard).

#### What is the current behavior?
When I try to create material-dashboard it is created with .css instead of .scss and with .spec file

#### What are the steps to reproduce?
    ng new <project-name> --style=scss --skip-tests
    ng add @angular/material@6.2.1 <-- I use 6.2.1 because https://github.com/angular/material2/issues/11836
    ng generate @angular/material:material-dashboard --name dashboard

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
    "@angular/animations": "^6.0.0",
    "@angular/cdk": "^6.2.0",
    "@angular/common": "^6.0.0",
    "@angular/compiler": "^6.0.0",
    "@angular/core": "^6.0.0",
    "@angular/forms": "^6.0.0",
    "@angular/http": "^6.0.0",
    "@angular/material": "^6.2.1",
    "@angular/platform-browser": "^6.0.0",
    "@angular/platform-browser-dynamic": "^6.0.0",
    "@angular/router": "^6.0.0",

    "@angular/cli": "~6.0.1",

#### Is there anything else we should know?

I see that command "ng add @angular/material" add only theme file in angular.json 
```
"styles": [
    {
        "input": "node_modules/@angular/material/prebuilt-themes/indigo-pink.css"
    },
    "src/styles.scss"
],
```
May be that command has to add schematic too?
```
"schematics": {
    "@schematics/material:material-dashboard": {
        "styleext": "scss",
        "spec": false
    },
    ....
}
```

### PR DESCRIPTION
**Pull request by DevVersion**
If developers create a new Angular CLI project, options like `--skipTests`, `--inlineTemplate` or `--inlineStyle` can be specified. These options will be stored in the workspace config file and will be respected if someone generates another component on the existing project.

Since we technically also generate components, we should respect the default option value for those options if _not_ explicitly specified.

Closes #11874.

### CODE REVIEW

File: src/lib/schematics/utils/build-component.ts [hunk 1]
```diff
@@ -168,9 +168,7 @@ export function buildComponent(options: ComponentOptions,
     const schematicFilesPath = resolve(dirname(context.schematic.description.path),
         schematicFilesUrl);
 
-    if (!options.styleext) {
-      options.styleext = determineDefaultStyleExt(project);
-    }
+    setDefaultSchematicOptions(project, options);
```
Comment (DevVersion) [inline src/lib/schematics/utils/build-component.ts]: @jelbourn I'm very unhappy with that function name. Do you have any better ideas?

File: src/lib/schematics/utils/schematic-options.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * Gets the default value for the specified option. The default options will be determined
+ * by looking at the stored schematic options for `@schematics/angular:component` in the
+ * CLI workspace configuration.
+ */
+function getDefaultSchematicOption<T>(project: WorkspaceProject, optionName: string,
+                                      fallbackValue: T): T | null {
+  if (project.schematics &&
+      project.schematics['@schematics/angular:component'] &&
+      project.schematics['@schematics/angular:component'][optionName] !== undefined) {
```
Comment (jelbourn) [inline src/lib/schematics/utils/schematic-options.ts]: `!= null`?

File: src/lib/schematics/utils/build-component.ts (hunk 1 above)
Comment (jelbourn) [reply src/lib/schematics/utils/build-component.ts]: How about
```ts
options = {...getDefaultComponentOptions(project), ...options};
```

File: src/lib/schematics/utils/schematic-options.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    options.spec = getDefaultSchematicOption(project, 'spec', true);
+  }
+}
+
+/**
+ * Gets the default value for the specified option. The default options will be determined
+ * by looking at the stored schematic options for `@schematics/angular:component` in the
+ * CLI workspace configuration.
+ */
+function getDefaultSchematicOption<T>(project: WorkspaceProject, optionName: string,
```
Comment (jelbourn) [inline src/lib/schematics/utils/schematic-options.ts]: `getDefaultComponentOption`?

File: src/lib/schematics/utils/build-component.ts (hunk 1 above)
Comment (DevVersion) [reply src/lib/schematics/utils/build-component.ts]: That would have been perfect. Unfortunately the options will look something like: `{spec: undefined}` and therefore wouldn't be replaced by the default component options.

I've went with something like that:

```ts

// Add the default component option values to the options if an option is not explicitly
// specified but a default component option is available.
Object.keys(options)
  .filter(optionName => options[optionName] == null && defaultComponentOptions[optionName])
  .forEach(optionName => options[optionName] = defaultComponentOptions[optionName]);
```

File: src/lib/schematics/utils/schematic-options.ts (hunk 2 above)
Comment (DevVersion) [reply src/lib/schematics/utils/schematic-options.ts]: Done. Please have another look.

**Pull request merged**